e2e/actions.go

```diff
@@ -0,0 +1,47 @@
+package e2e
+
+import (
+	"log"
+	"math/rand"
+	"sync"
+
+	"github.com/0xPolygon/pbft-consensus"
+)
+
+type RevertFunc func()
+
+type FunctionalAction interface {
+	CanApply(c *Cluster) bool
+	Apply(c *Cluster) RevertFunc
+}
+
+// Encapsulates logic for dropping nodes action.
+type DropNodeAction struct {
+	lock sync.Mutex
```

Comment:
Why do we need this lock? Once the cluster has started (`Start()` called) this is the only place where `Start` and `Stop` are going to be called and they are already synced over the framework with the revert and apply channels.

Reply:
In fact apparently we don't need it eventually. Although at some point you have suggested it (but now seems to me that you were referring to the logic contained within `Start` function: https://github.com/0xPolygon/pbft-consensus/pull/24#discussion_r810993635) and I thought that maybe we would spawn some additional routines for apply, revert and/or cluster validation, but seems that wouldn't be the case at the end, so we can remove lock.